diff/parse.go

```diff
@@ -223,8 +230,31 @@ func (r *FileDiffReader) HunksReader() *HunksReader {
 
 // ReadFileHeaders reads the unified file diff header (the lines that
 // start with "---" and "+++" with the orig/new file names and
-// timestamps).
+// timestamps). Or starts with "Only in " message with dir path filename.
 func (r *FileDiffReader) ReadFileHeaders() (origName, newName string, origTimestamp, newTimestamp *time.Time, err error) {
+	if (r.fileHeaderLine != nil) && (bytes.HasPrefix(r.fileHeaderLine, onlyInMessagePrefix)) {
+		path := bytes.Split(bytes.TrimPrefix(
```

Comment:
I think regexp matching with Submatches would be much clearer here. Alternatively you can try one of the fmt.Scanf functions.